sources/engine/Xenko/Rendering/ParameterCollection.cs

```diff
@@ -317,6 +317,35 @@ public T Get<T>(PermutationParameterKey<T> parameter, bool createIfNew = false)
             return values;
         }
 
+        /// <summary>
+        /// Sets blittable values by copying a <see cref="ParameterCollection"/>'s values to this collection.
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="key">The key to store the copied values.</param>
+        /// <param name="source">The collection to retrieve the values to copy.</param>
+        /// <param name="sourceKey">The key for the source collection for the values to copy.</param>
+        public unsafe void SetValues<T>(ValueParameterKey<T> key, ParameterCollection source, ValueParameterKey<T> sourceKey) where T : struct
```

Comment:
Trying to think about a better name. Maybe something like `CopyValues` or `CopyValuesFrom`?

Reply:
If I change to use 'Copy', would you prefer `CopyTo` rather than 'from'?
A quick glance over the codebase, the standard .NET collections/arrays uses 'to' rather than 'from'.
Though this will also make this single line stick out:
```
diffuseParameters.CopyTo(SphericalHarmonicsEnvironmentColorKeys.SphericalColors, parameters, sphericalColorsKey);
parameters.Set(specularCubeMapkey, specularCubemap);
parameters.Set(specularMipCountKey, specularCubemapLevels);
```
(Will also need advice on the preferred parameter ordering!)

Reply:
Yes, definitely!
At first I was suggesting a name to keep current signature, but I agree with you that reversing ownership and changing method with `CopyTo` and a destination is probably a much better fit with typical API (I was also thinking about it).
Let's go for that then.

Reply:
Ok, done.